app/main.py

```diff
@@ -1,3 +1,78 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance | int | float") -> "Distance":
+        if isinstance(other, Distance):
+            new_distance = self.km + other.km
+        elif isinstance(other, (int, float)):
+            new_distance = self.km + other
+        else:
+            raise TypeError
+        return Distance(new_distance)
```

Comment:
1. We don't need to raise Error in this case
2. Don't do 2 isinstance method in this case, just use `else`
3. Don't use `""` for your type hints, import `from __future__ import annotations`, if you need to use more advanced annotations and your python version is lower then you need

Good example:
`def __add__(self, other: Distance | int | float) -> Distance:`
Fix all of this in all methods